### ISSUE TITLE
Configurable Interface Font/Font Size

### ISSUE THREAD

**Issue by dnjp** [zed-industries/feedback#330]
**Describe the solution you'd like**
I would like to be able to configure both the font and font size for the interface surrounding the editor window. Currently, the default font does not match the font or font size I'm using for the editor window.

<img width="872" alt="image" src="https://user-images.githubusercontent.com/37786372/179285157-0f7236d6-4562-4d4a-b03f-92bed4b5ea5b.png">

**Comment by alevroub** [comment]
Another suggestion could be to simply use the macOS system font for the UI. Iosevka looks great for the most part, but I find the double-story lowercase `g`, as well as the terminals of the lowercase `t` and `f` to be too distracting for UI use.

**Comment by jansol** [comment]
IIRC macOS does not expose the system UI font to applications directly, only through the Cocoa/UIKit widgets. It would of course be possible to hardcode a path to right file in the system directory but there are no guarantees that it won't change without a warning. And simply shipping a copy of the font file with applications is likely not allowed.

**Comment by TheFoolishPupil** [comment]
Yes this is a great to have, not just the font size but the scale in general. I use a mbp16 and look how jarring the scale difference between zed and sublime is in my case. 

<img width="1074" alt="Screen Shot 2022-08-16 at 12 43 15" src="https://user-images.githubusercontent.com/43202406/184860881-760fb387-67a2-4551-8288-1f5243f3cb05.png">

**Comment by chriskrycho** [comment]
Apple actually [supplies the fonts directly](https://developer.apple.com/fonts/); the key would be checking the license for distribution.

**Comment by jansol** [comment]
The license page of the installer says no:
> IMPORTANT NOTE:  THE APPLE SF MONO FONT IS TO BE USED SOLELY FOR CREATING MOCK-UPS OF USER INTERFACES TO BE USED IN SOFTWARE PRODUCTS RUNNING ON APPLE’S iOS, iPadOS, macOS, tvOS OR watchOS OPERATING SYSTEMS, AS APPLICABLE.

**Comment by chriskrycho** [comment]
Ah, drat.

**Comment by jansol** [comment]
So far it seems the only official way to access those fonts is with [`CTFontCreateUIFontForLanguage(_:_:_:)`](https://developer.apple.com/documentation/coretext/1511312-ctfontcreateuifontforlanguage) and then I guess asking CoreText or CoreGraphics for a rasterized glyph bitmap or a CGPath from which it may be possible to extract the curve data.

If pre-rasterized glyphs are enough I guess this is fine although it requires an entire custom code path for a single font. Ultimately it seems like a lot of effort for relatively little gain.

However it looks like fetching the base *metrics* of the system font would be quite straightforward. Those could be used to pick a default size (at runtime) for the shipped font that is as close as possible to the configured system font size. This could then be reused to honor the system font size defaults on platforms that allow changing them. (Windows and Linux do, as do iOS and Android IIRC)

**Comment by sonereker** [comment]
How about Inter font? https://rsms.me/inter/

I think JetBrains uses this one by default now.

**Comment by alevroub** [comment]
@sonereker there's no reason to use Inter when San Francisco is already available in macOS (and also is a much better drawn typeface)

**Comment by trojanfoe** [comment]
Is it possible to simply add a setting for the UI font and size and let the user worry about supplying the actual font?
